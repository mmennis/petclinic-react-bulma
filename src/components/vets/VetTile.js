import React from 'react'
import PropTypes from 'prop-types'
import { Tile, Card, Content, Heading, Button } from 'react-bulma-components'
import EditVetForm from './EditVetForm'
import axios from 'axios'
import { data } from './vets.localisation.data'
import LocalizedStrings from 'react-localization'
import { LanguageContext } from '../localization/LanguageContext'

const strings = new LocalizedStrings(data)

const BASE_URL = process.env.REACT_APP_PETCLINIC_APP_API_ENDPOINT;

export default class VetTile extends React.Component {

    constructor(props) {
        super(props)

        this.state = {
            updated: false
        }
        this.handleVetUpdate = this.handleVetUpdate.bind(this)
    }

    handleVetUpdate = (newVet) => {
        console.log(`Received updated vet data ${JSON.stringify(newVet)}`)
        axios.put(`${BASE_URL}/vets/${newVet._id}`, newVet)
            .then((response) => {
                console.log(`Response for vet update: ${response.data.msg}`)
            })
            .catch((err) => {
                console.error(`Problem with Vet update request: ${err}`)
            }).finally(() => {
                console.log('Update vet complete')
            })
    }

    handleDeleteVet = () => {
        this.props.handleDeleteVet(this.props.vet)
    }

    capitalize = (word) => {
        return word.charAt(0).toUpperCase() + word.substring(1)
    }

    render() {
        const localeContext = this.context
        strings.setLanguage(localeContext.lang)
        return (
            <Tile size={3}>
                <Card paddingless={true} rounded="true" outlined="true">
                    <Card.Header outlined="true" >
                        <Card.Header.Title>{this.props.vet.first_name} {this.props.vet.last_name}</Card.Header.Title>
                    </Card.Header>
                    <Card.Content paddingless={false} color="light">
                        <Content size="medium">
                            <p>{this.capitalize(this.props.vet.specialty)}</p>
                        </Content>
                        <Content>
                            <Heading marginless={true} paddingless={false} size={6}>{strings.address_heading}</Heading>
                            <p>{this.props.vet.address}<br/>{this.props.vet.city}, {this.props.vet.state}</p>
                        </Content>
                        <Content size="small">
                            <p>{this.props.vet.telephone}</p>
                        </Content>
                        <Content size="small">
                            <p>{strings.hours}: {this.props.vet.office_hours}</p>
                        </Content>

                    </Card.Content>
                    <Card.Footer>
                        <EditVetForm 
                            vet={this.props.vet} 
                            handleVetUpdate={this.handleVetUpdate}
                            modal={{closeOnBlur: true, showClose: true }}
                        />
                        <Button 
                            onClick={this.handleDeleteVet} 
                            size="small" color="danger"
                            style={{ 'marginLeft': '10px', 'marginRight': '10px' }}>
                            {strings.delete_button}
                        </Button>
                    </Card.Footer>
                </Card>
            </Tile>
        )
    }
}

VetTile.contextType = LanguageContext

VetTile.propTypes = {
    vet: PropTypes.object.isRequired,
    handleDeleteVet: PropTypes.func.isRequired,
}