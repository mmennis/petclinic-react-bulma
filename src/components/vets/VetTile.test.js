import React from 'react'
import { create } from 'react-test-renderer'
import { render } from '@testing-library/react'
import VetTile from './VetTile'

const testVet = {
    first_name: 'First',
    last_name: 'Last',
    address: '1234 Main St',
    city: 'Cupertino',
    state: 'CA',
    telephone: '(408) 555-1212',
    office_hours: '8:00 AM - 5:00 PM'
}

describe('VetTile', () => {

    it('should render as per the snapshot', () => {
        const vetTile = create(<VetTile vet={testVet}/>)
        expect(vetTile.toJSON()).toMatchSnapshot()
    })

    it('should have vet details in the output', () => {
        console.error = jest.fn()
        const { getByText } = render(<VetTile vet={testVet} />)
        const vetFName = getByText(/First/i)
        expect(vetFName).toBeInTheDocument();
        const vetLName = getByText(/Last/i)
        expect(vetLName).toBeInTheDocument()
        const vetHours = getByText(/office/i)
        expect(vetHours).toBeInTheDocument()
    })

    it('should log an error to the console if required owner props missing', () => {
        console.error = jest.fn()
        React.createElement(VetTile)
        expect(console.error).toHaveBeenCalledTimes(1)
    })

})