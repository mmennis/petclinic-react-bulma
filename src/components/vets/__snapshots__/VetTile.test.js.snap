// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VetTile should render as per the snapshot 1`] = `
<div
  className="tile is-3"
>
  <div
    className="card is-paddingless"
    outlined="true"
    rounded="true"
  >
    <div
      className="card-header"
      outlined="true"
    >
      <div
        className="card-header-title"
      >
        First
         
        Last
      </div>
    </div>
    <div
      className="card-content"
      color="light"
    >
      <div
        className="content is-medium"
      >
        <p />
      </div>
      <div
        className="content"
      >
        <h1
          className="title is-6 is-marginless"
        >
          Address
        </h1>
        <p>
          1234 Main St
          <br />
          Cupertino
          , 
          CA
        </p>
      </div>
      <div
        className="content is-small"
      >
        <p>
          (408) 555-1212
        </p>
      </div>
      <div
        className="content is-small"
      >
        <p>
          Hours: 
          8:00 AM - 5:00 PM
        </p>
      </div>
    </div>
    <div
      className="card-footer"
    >
      <div>
        <button
          className="is-primary is-small button"
          data-testid="modal-open"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "20px",
              "marginRight": "10px",
            }
          }
          tabIndex={0}
        >
          Update
        </button>
        <form
          data-testid="form"
          onSubmit={[Function]}
        />
      </div>
      <button
        className="is-danger is-small button"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "10px",
            "marginRight": "10px",
          }
        }
        tabIndex={0}
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
